Description: Allow TPM2 key files
Author: David Woodhouse <dwmw2@infradead.org>
Origin: upstream, https://gitlab.gnome.org/GNOME/NetworkManager-openconnect/commit/9b3be7fc510b469af293f6d15e1bb5cadc1adca5
Last-Update: 2018-10-13
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/properties/auth-helpers.c
+++ b/properties/auth-helpers.c
@@ -166,6 +166,8 @@
 static const char *pem_key_begin = "-----BEGIN PRIVATE KEY-----";
 static const char *pem_enc_key_begin = "-----BEGIN ENCRYPTED PRIVATE KEY-----";
 static const char *pem_tss_keyblob_begin = "-----BEGIN TSS KEY BLOB-----";
+static const char *pem_tss2_keyblob_begin = "-----BEGIN TSS2 KEY BLOB-----";
+static const char *pem_tss2_privkey_begin = "-----BEGIN TSS2 PRIVATE KEY-----";
 static const char *pem_cert_begin = "-----BEGIN CERTIFICATE-----";
 
 static gboolean
@@ -229,6 +231,16 @@
 		show = TRUE;
 		goto out;
 	}
+
+	if (find_tag (pem_tss2_keyblob_begin, (const char *) contents, bytes_read)) {
+		show = TRUE;
+		goto out;
+	}
+
+	if (find_tag (pem_tss2_privkey_begin, (const char *) contents, bytes_read)) {
+		show = TRUE;
+		goto out;
+	}
 
 	if (find_tag (pem_cert_begin, (const char *) contents, bytes_read)) {
 		show = TRUE;
